src/components/OSCALControlProse.js

```diff
@@ -259,27 +272,159 @@ export function OSCALReplacedProseWithParameterLabel(props) {
     );
   }
 
+  let labelWithProse = props.prose;
+  if (props.label) {
+    labelWithProse = props.label.concat(` ${props.prose}`);
+  }
+
+  const partialRestData = cloneDeep(props.implementedRequirement);
+  const statement =
+    partialRestData?.statements?.find(
+      (element) => element["statement-id"] === props.statementId
+    ) || null;
+  const paramId = props.prose.split(
+    RegExp(prosePlaceholderRegexpString, "g")
+  )[1];
+
   return (
     <Typography className={props.className}>
-      {props.label}
-      {props.prose
+      {labelWithProse
         .split(RegExp(prosePlaceholderRegexpString, "g"))
         .map((segment, index) => {
           if (index % 2 === 0) {
             return getTextSegment(segment, index.toString());
           }
-          return getParameterLabelSegment(
-            props.parameters,
-            segment,
-            props.modificationSetParameters,
-            index.toString()
+
+          return (
+            <Typography display="inline" variant="body2">
+              {getParameterLabelSegment(
+                props.parameters,
+                segment,
+                props.modificationSetParameters,
+                index.toString()
+              )}
+            </Typography>
           );
         })}
       {props.modificationDisplay}
+      {props.label && props.isEditable ? (
+        <IconButton
+          onClick={(event) => {
+            setAnchorEl(event.currentTarget);
+          }}
+        >
+          <EditIcon fontSize="small" />
+        </IconButton>
+      ) : null}
+      {props.isEditable ? (
+        <Grid container>
+          <OSCALPopover
+            anchorEl={anchorEl}
+            setAnchorEl={setAnchorEl}
+            controlId={props.controlId}
+            onCancel={() => {
+              setAnchorEl(null);
+            }}
+            onFieldSave={(descriptionReference, implementationReference) => {
```

Comment:
This is definitely too complicated to declare inline. Can we define it at least at the top of the function if not on its own entirely.